frontend/packages/dev-console/console-extensions.json

```diff
@@ -685,6 +685,83 @@
       "insertAfter": "console.preferredCreateEditMethod"
     }
   },
+  {
+    "type": "console.user-preference/item",
+    "properties": {
+      "id": "devconsole.preferredResource",
+      "label": "%devconsole~Resource type%",
+      "groupId": "applications",
+      "description": "%devconsole~If resource type is not selected, the console default to the latest.%",
+      "field": {
+        "type": "dropdown",
+        "userSettingsKey": "devconsole.preferredResourceType",
+        "defaultValue": "kubernetes",
+        "description": "%devconsole~The defaults below will only apply to the Import from Git and Deploy Image forms when creating Deployments or Deployment Configs.%",
+        "options": [
+          {
+            "value": "latest",
+            "label": "Last used",
+            "description": "%devconsole~The last resource type used when adding a resource.%"
+          },
+          {
+            "value": "kubernetes",
+            "label": "%devconsole~Deployment%",
+            "description": "%devconsole~A Deployment enables declarative updates for Pods and ReplicaSets.%"
+          },
+          {
+            "value": "openshift",
+            "label": "%devconsole~DeploymentConfig%",
+            "description": "%devconsole~A DeploymentConfig defines the template for a Pod and manages deploying new Images or configuration changes.%"
+          }
+        ]
+      },
+      "insertBefore": "devconsole.routingOptions"
+    },
+    "flags": {
+      "disallowed": ["KNATIVE_SERVING"]
```

Comment:
shouldn't this be `FLAG_KNATIVE_SERVING_SERVICE` as we have here https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/import/section/ResourceSection.tsx#L73 for resource selection?

as now i can see serverless deployment in user preferences though have not created CR for knativeServing yet/.

Reply:
used `KNATIVE_SERVING_SERVICE` flag here.